### PR TITLE
Merge main from dependabot/dependabot-core

### PR DESCRIPTION
**Pull request by GiriB**
rebase the forked repository with dependaot/dependabot-core:main branch

### CODE REVIEW

File: npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker.rb [hunk 1]
```diff
@@ -297,9 +297,8 @@ def library?
 
       def dependency_source_details
         sources =
-          dependency.requirements.map { |r| r.fetch(:source) }.uniq.compact
-
-        raise "Multiple sources! #{sources.join(', ')}" if sources.count > 1
+          dependency.requirements.map { |r| r.fetch(:source) }.uniq.compact.
+          sort_by { |source| RegistryFinder.central_registry?(source[:url]) ? 1 : 0 }
```
Comment (GiriB) [inline npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker.rb:301]: Should not effect us right?

File: Dockerfile [hunk 2]
```diff
@@ -70,7 +70,7 @@ RUN apt-add-repository ppa:brightbox/ruby-ng \
   && apt-get install -y ruby2.6 ruby2.6-dev \
   && gem update --system 3.2.14 \
   && gem install bundler -v 1.17.3 --no-document \
-  && gem install bundler -v 2.2.15 --no-document \
+  && gem install bundler -v 2.2.18 --no-document \
```
Comment (milind009) [inline Dockerfile:73]: @AlekhyaYalla we will have to make a similar change in our Dockerfile.lite

File: common/lib/dependabot/config.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+module Dependabot
+  module Config
+    class InvalidConfigError < StandardError; end
```
Comment (milind009) [inline common/lib/dependabot/config.rb:5]: nit: we could use something like this for tracking errors due to incorrect checkin in dmaas.yml

Comment (milind009) [reply common/lib/dependabot/config.rb:5]: Not essentially in this PR but something to keep in mind

**Review by milind009 — APPROVED** [review]

**Pull request merged**